### PR TITLE
test: add selected logprob parity harness

### PR DESCRIPTION
**Pull request by Txy02**
## Summary

Adds a small selected-logprob parity harness for validating that the same effective completion rows produce identical selected logprobs across batch layout changes.

This is intended as a first step toward the rollout-vs-training numerical parity work discussed in #148, #152, and #154.

## What is covered

- Batch-position invariance for selected logprobs
- Padding-layout invariance for selected logprobs
- Drift summaries through the existing `summarize_kernel_drift`
- Optional CUDA FP32/BF16 dtype sweep when CUDA is available

## What is intentionally out of scope

- No kernel implementation changes
- No vLLM integration yet
- No benchmark changes
- No external framework integration

## Tests

Validated on AutoDL RTX 4090:

- `python -m pytest tests/test_logprob_parity.py -v`
- `python -m pytest tests/test_reference_ops.py tests/test_logprob_parity.py -v`

Result:

- `4 passed`
- `11 passed`

## Summary by CodeRabbit

* **New Features**
  * Added testing helpers to generate padded batch layouts and compare selected log-probability results across different batch/padding layouts.
* **Tests**
  * Added invariance tests for batch position and padding layout.
  * Added negative tests for invalid row mappings and out-of-range padding token IDs.
  * Added CUDA dtype sweep coverage for `float32` and `bfloat16`, validating consistent parity in a chosen output dtype.
* **Chores**
  * Re-exported the new testing helpers from the main testing utilities entry point.

### CODE REVIEW

**Comment by Txy02** [comment]
Fixed the pre-commit EOF issue and re-ran local checks:

- `pre-commit run --files rl_engine/testing/logprob_parity.py tests/test_logprob_parity.py rl_engine/testing/__init__.py`
- `python -m pytest tests/test_logprob_parity.py -v` (`9 passed`)
- `python -m pytest tests/test_reference_ops.py tests/test_logprob_parity.py -v` (`16 passed`)

**Comment by Flink-ddd** [comment]
cc @maxiaosong1124 @a-kaa PTAL

**Review by KJLdefeated — COMMENTED** [review]
Overall Good to me. One thing before this is a trustworthy parity harness: every test that measures drift asserts exactly zero, and the two negative tests only cover input validation, so the measurement path (summarize_kernel_drift + the remap) is never shown to register a real difference. As-is, a silently-broken detector that always returned zero would pass the whole suite. Add a positive control so the zero-drift assertions mean something:
```python
def test_compare_detects_drift_when_candidate_differs():
    """Positive control: a genuinely different candidate must register non-zero
    drift, so the zero-drift tests above can't pass merely by the measurement
    path always returning zero."""
    logits, token_ids, mask = _case()
    candidate_logits = logits.clone()
    candidate_logits[0, 0, 0] += 5.0  # bump one vocab logit at an active position -> logsumexp shifts -> selected logprob changes
    summary = compare_selected_logprob_layouts(
        logits, token_ids, mask,
        candidate_logits, token_ids, mask,
        candidate_rows=torch.arange(logits.shape[0]),
    )
    assert summary["max_abs_error"] > 0.0
```
